src/librustc/middle/stability.rs

```diff
@@ -597,8 +597,31 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                                            feature.as_str(), &r),
                     None => format!("use of unstable library feature '{}'", &feature)
                 };
-                emit_feature_err(&self.sess.parse_sess, &feature.as_str(), span,
-                                 GateIssue::Library(Some(issue)), &msg);
+
+
+                let msp: MultiSpan = span.into();
+                let cm = &self.sess.parse_sess.codemap();
+                let span_key =
+                    msp.primary_span().and_then(|sp:Span|
+                        if sp != DUMMY_SP {
+                            let fname = cm.lookup_char_pos(sp.lo()).file.as_ref().name.clone();
+                            if fname.starts_with("<") && fname.ends_with(" macros>") {
+                                None
+                            } else {
+                                Some(span)
+                            }
+                        } else {
+                            None
+                        }
+                    );
+
+                let tuple = (None, span_key, msg.clone());
```

Comment:
Try to give `tuple` a more descriptive name. You can also get around that by just inserting the tuple inline ;)